packages/api/src/services/lsp-state.e2e.ts

```diff
@@ -0,0 +1,81 @@
+require("dotenv").config();
+import * as uma from "@uma/sdk";
+import assert from "assert";
+import { ethers } from "ethers";
+import Service from "./lsp-state";
+import type { AppState } from "../";
+import { Multicall } from "@uma/sdk";
+import { lsps } from "../tables";
+// this fixes usage of "this" as any
+import type Mocha from "mocha";
+
+type Dependencies = Pick<
+  AppState,
+  "lsps" | "registeredLsps" | "provider" | "collateralAddresses" | "shortAddresses" | "longAddresses" | "multicall"
+>;
+
+const registeredContracts = ["0x372802d8A2D69bB43872a1AABe2bd403a0FafA1F"];
+
+describe("lsp-state service", function () {
+  let appState: Dependencies;
+  before(async function () {
+    assert(process.env.CUSTOM_NODE_URL);
+    assert(process.env.multicallAddress);
+    const provider = new ethers.providers.WebSocketProvider(process.env.CUSTOM_NODE_URL);
+    appState = {
+      provider,
+      multicall: new Multicall(process.env.multicallAddress, provider),
+      registeredLsps: new Set<string>(registeredContracts),
+      collateralAddresses: new Set<string>(),
+      longAddresses: new Set<string>(),
+      shortAddresses: new Set<string>(),
+      lsps: {
+        active: lsps.JsMap("Active LSP"),
+        expired: lsps.JsMap("Expired LSP"),
+      },
+    };
+  });
+  it("init", async function () {
+    const service = Service(undefined, appState);
+    assert.ok(service);
+  });
+  it("readDynamicState", async function () {
+    const [address] = registeredContracts;
+    const service = Service(undefined, appState);
+    const instance = await uma.clients.lsp.connect(address, appState.provider);
+    const result = await service.utils.batchRead(service.utils.dynamicProps, instance, address);
+    assert.ok(result.address);
+    assert.ok(result.updated);
+    assert.ok(result.expiryPrice);
+    assert.ok(result.expiryPercentLong);
+    assert.ok(result.expirationTimestamp);
+    assert.ok(result.contractState >= 0);
+  });
+  it("readStaticState", async function () {
+    const [address] = registeredContracts;
+    const service = Service(undefined, appState);
+    const instance = await uma.clients.lsp.connect(address, appState.provider);
+    const result = await service.utils.batchRead(service.utils.staticProps, instance, address);
```

Comment:
these tests do do any validation on the _data_, and just check they are not null/are ok. do you see any merit in validating the correctness of the data returned?

Reply:
thats interesting, havent thought of that.  i might need convincing, like if theres a particular scenario where we would need to do this for a piece of data. my thoughts in general are that we want as dumb a pass through to the contract as possible  and assume we can trust the values from it.  otherwise though i think it could be fairly easy to add validation on a per prop basis if you think we need it.

Reply:
oh i just realized this is in the test, to validate the values we get back. hmm yea we could, though some of the dynamic props we might not want to for the test

Reply:
added checks against expected value